### ISSUE TITLE
No module named 'fairseq.data.data_utils_fast'

### ISSUE THREAD

**Issue by SuHe36** [pytorch/fairseq#2939]
I want to finetune BART-large model, my script is:
```
TOTAL_NUM_UPDATES=20000
WARMUP_UPDATES=500
LR=3e-05
MAX_TOKENS=4000
UPDATE_FREQ=1
BART_PATH=../model/bart/bart.large/model.pt

CUDA_VISIBLE_DEVICES=1 nohup python ../train.py ../gec_data-bin \
    --save-dir ../model/gec_bart \
    --log-format simple \
    --log-interval 100 \
    --seed 1 \
    --restore-file $BART_PATH \
    --max-tokens $MAX_TOKENS \
    --task translation \
    --source-lang src --target-lang trg \
    --truncate-source \
    --layernorm-embedding \
    --share-all-embeddings \
    --share-decoder-input-output-embed \
    --reset-optimizer --reset-dataloader --reset-meters \
    --arch bart_large \
    --criterion label_smoothed_cross_entropy \
    --label-smoothing 0.1 \
    --dropout 0.3 --attention-dropout 0.3 \
    --weight-decay 0.01 --optimizer adam --adam-betas "(0.9, 0.999)" --adam-eps 1e-08 \
    --clip-norm 0.1 \
    --lr-scheduler polynomial_decay --lr $LR --total-num-update $TOTAL_NUM_UPDATES --warmup-updates $WARMUP_UPDATES \
    --skip-invalid-size-inputs-valid-test \
    --max-epoch 10 \
    --find-unused-parameters >> bart.log 2>&1
```

but I met the following error:
```
Traceback (most recent call last):
  File "/home/research/hesu/bart_gec/fairseq/data/data_utils.py", line 221, in batch_by_size
    import data_utils_fast
ModuleNotFoundError: No module named 'data_utils_fast'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "../train.py", line 11, in <module>
    cli_main()
  File "/home/research/hesu/bart_gec/fairseq_cli/train.py", line 307, in cli_main
    main(args)
  File "/home/research/hesu/bart_gec/fairseq_cli/train.py", line 83, in main
    extra_state, epoch_itr = checkpoint_utils.load_checkpoint(args, trainer)
  File "/home/research/hesu/bart_gec/fairseq/checkpoint_utils.py", line 153, in load_checkpoint
    epoch=0, load_dataset=True, **passthrough_args
  File "/home/research/hesu/bart_gec/fairseq/trainer.py", line 266, in get_train_iterator
    epoch=epoch,
  File "/home/research/hesu/bart_gec/fairseq/tasks/fairseq_task.py", line 158, in get_batch_iterator
    required_batch_size_multiple=required_batch_size_multiple,
  File "/home/research/hesu/bart_gec/fairseq/data/data_utils.py", line 225, in batch_by_size
    'Please build Cython components with: `pip install --editable .` '
ImportError: Please build Cython components with: `pip install --editable .` or `python setup.py build_ext --inplace`
```

Then I followed the instructions in the error report and did the following operation:
`pip install --editable .` and `python setup.py build_ext --inplace`

No error occurred.

I also try to import data_utils_fast in the terminal:
```
>>> import fairseq
>>> from fairseq.data.data_utils_fast import batch_by_size_fast
>>>
```
It imports successfully.

But when I re-execute the script, it still reports the same error.
```
ModuleNotFoundError: No module named 'data_utils_fast'
```

I tried to use `pip install fairseq` to install fairseq,  and use the following script to finetune the model. 
It can successfully run.
```
TOTAL_NUM_UPDATES=20000
WARMUP_UPDATES=500
LR=3e-05
MAX_TOKENS=4000
UPDATE_FREQ=1
BART_PATH=../model/bart/bart.large/model.pt

CUDA_VISIBLE_DEVICES=1 nohup python fairseq-train ../gec_data-bin \
    --save-dir ../model/gec_bart \
    --log-format simple \
    --log-interval 100 \
    --seed 1 \
    --restore-file $BART_PATH \
    --max-tokens $MAX_TOKENS \
    --task translation \
    --source-lang src --target-lang trg \
    --truncate-source \
    --layernorm-embedding \
    --share-all-embeddings \
    --share-decoder-input-output-embed \
    --reset-optimizer --reset-dataloader --reset-meters \
    --arch bart_large \
    --criterion label_smoothed_cross_entropy \
    --label-smoothing 0.1 \
    --dropout 0.3 --attention-dropout 0.3 \
    --weight-decay 0.01 --optimizer adam --adam-betas "(0.9, 0.999)" --adam-eps 1e-08 \
    --clip-norm 0.1 \
    --lr-scheduler polynomial_decay --lr $LR --total-num-update $TOTAL_NUM_UPDATES --warmup-updates $WARMUP_UPDATES \
    --skip-invalid-size-inputs-valid-test \
    --max-epoch 10 \
    --find-unused-parameters >> bart.log 2>&1
```

**So now we can't  install fairseq  from source, or it just a bug when we use fairseq to finetune BART model ?** 

My environment is:

- fairseq Version: 0.9.0/0.10.0
- pytorch version = 1.4.0
- python version = 3.7.0
- cuda version = 9.2/10.0.130

**Comment by myleott** [comment]
It's not clear to me what version of fairseq you're using, since the traceback doesn't look like it matches the current master.

Your traceback:
```
Traceback (most recent call last):
  File "/home/research/hesu/bart_gec/fairseq/data/data_utils.py", line 221, in batch_by_size
    import data_utils_fast
ModuleNotFoundError: No module named 'data_utils_fast'
```

The current code in master:
https://github.com/pytorch/fairseq/blob/74fc8ccce1cfb7367cb57e6605e125030f2b0d31/fairseq/data/data_utils.py#L302-L311

Can you clarify what commit of fairseq you're using?

**Comment by SuHe36** [comment]
I tried to use fairseq 0.9.0 version and 0.10.0 version, both encountered this problem

**Comment by myleott** [comment]
Where does `/home/research/hesu/bart_gec/fairseq/data/data_utils.py` come from? The traceback you posted shows `import data_utils_fast`, which is not how it's implemented in either 0.9.0 or 0.10.0. It seems you've made some local changes to the code; please revert these.

In fairseq 0.9.0 the code looks like: https://github.com/pytorch/fairseq/blob/df2f84ce619edddb80e720a56abc74d5490fed99/fairseq/data/data_utils.py#L216-L222

In fairseq 0.10.0 the code looks like: https://github.com/pytorch/fairseq/blob/3e5a2739bce92b2516d060959a1aa3dd51d221dc/fairseq/data/data_utils.py#L302-L311
